collagraph/renderers/pygfx_renderer.py

```diff
@@ -39,42 +25,21 @@ def insert(
     def remove(self, el: gfx.WorldObject, parent: gfx.WorldObject):
         parent.remove(el)
 
-    def create_element(self, type: str) -> gfx.WorldObject:
-        """Create pygfx element for the given type"""
-        if type == "Point":
-            # NOTE: geometry and material should preferably be passed through
-            # state instead of generated here, because they are resources that
-            # can be shared
-            obj = MeshEvents(gfx.sphere_geometry(), gfx.MeshPhongMaterial())
-        if type == "Group":
-            obj = gfx.Group()
-        if type == "Mesh":
-            obj = MeshEvents()
-        return obj
-
     def set_attribute(self, obj, attr, value):
-        if attr == "color":
-            obj.material.color = value
-            return
-
-        if hasattr(value, "__len__") and len(value) == 3:
+        if attr == "position" or attr == "scale":
             value = gfx.linalg.Vector3(*value)
+        elif attr == "rotation":
+            value = gfx.linalg.Quaternion(*value)
+        elif attr == "matrix":
+            value = gfx.linalg.Matrix4(*value)
```

Comment:
I'm expecting this to break down at some point, but if it works for now, sure

Reply:
Yes, this is highly dependent on the pygfx API, so we should make sure to keep this in sync.

Reply:
I put a suggestion in a separate comment to explain better what I mean

Reply:
Ah, I might not have seen your edited comment :) It's clear now though: great suggestion!